projects/media-viewer/src/lib/media-viewer.component.ts

```diff
@@ -47,6 +48,7 @@ export class MediaViewerComponent implements OnChanges, AfterContentInit {
   constructor(
     public readonly toolbarButtons: ToolbarButtonVisibilityService,
     public readonly toolbarEvents: ToolbarEventService,
+    public readonly keyListenerService: KeyListenerService,
```

Comment:
I'm wondering whether it would be better to use a directive for this, so that even the host-listener stuff can be contained within that? just a thought, not sure if it serves the purpose better, but what it would give us is the whole functionality contained within the directive, and we just tag the media-viewer component with it to enable it

Reply:
It's worth a try.

Reply:
The listener is attached to the top component. But perhaps not all the bindings should be handled globally - for example in case of a modal or something similar...